simln-lib/src/lib.rs

```diff
@@ -1351,7 +1341,7 @@ async fn produce_simulation_results(
     results: Sender<(Payment, PaymentResult)>,
     listener: Listener,
 ) -> Result<(), SimulationError> {
-    let mut set = tokio::task::JoinSet::new();
+    let set = TaskTracker::new();
```

Comment:
I think that we can pass the top level task set in here as well rather than having a new set?

This is pre-existing, sorry I didn't mention it earlier!